packages/fxa-auth-server/fxa-oauth-server/lib/grant.js

```diff
@@ -222,5 +223,19 @@ exports.generateAccessToken = async function generateAccessToken(grant) {
     return accessToken;
   }
 
+  // This is awful. The auth server is the canonical source of truth
```

Comment:
I agree with your assessment, but I haven't been able to come up with a better suggestion, especially since we need to profile info to update to reflect the current state of the user's account when using the refreshToken grant.